lib/puma/signal.rb

```diff
@@ -0,0 +1,42 @@
+module Puma
+  module Signal
+    module_function
+
+    def prepend_handler(sig, &handler)
+      name = signame(sig)
+      custom_signal_handlers[signame] ||= []
+      custom_signal_handlers[signame].prepend(handler)
+    end
```

Comment:
wonder if we shouldn't also add `append_handler`? Does it make sense?

Reply:
Yeah it makes sense. `prepend` initially had meaning like `prepend before Puma handlers`, so yeah, I'd add this method as well.

Reply:
We can add when needed 👍